src/redux/books/books.js

```diff
@@ -37,8 +16,21 @@ export const bookReducer = (state = initialdata, action) => {
       },
     ];
   }
+  if (action.type === GET_BOOKS.type) {
+    return action.payload;
+  }
   if (action.type === REMOVE_BOOK.type) {
     return state.filter((item) => item.id !== action.payload);
   }
   return state;
 };
+
+export const removeBook = (bookId) => async (dispatch) => {
+  api.deleteBook(bookId);
+  dispatch(REMOVE_BOOK(bookId));
+};
+
+export const getBooks = () => async (dispatch) => {
+  const books = await api.fetchBooks();
+  dispatch(GET_BOOKS(books));
+};
```

Comment:
- [ ] @RashidMenhas great work on your reducer however it's a project requirement to use `createAsyncThunk`, kindly ensure your reducer users it.
![image](https://user-images.githubusercontent.com/61408860/204048514-88b60e3c-4b7a-4145-856d-bd21e96d2dd2.png)

Reply:
@RashidMenhas as previously requested kindly use `createAsyncThunk` its a project requirement.
``example using it in a code``
```
export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },

```